sway/commands/input/xkb_switch_layout.c

```diff
@@ -100,6 +100,10 @@ struct cmd_results *input_cmd_xkb_switch_layout(int argc, char **argv) {
 
 		action->keyboard = wlr_keyboard_from_input_device(dev->wlr_device);
 		if (relative) {
+			if (action->keyboard->keymap == NULL && dev->is_virtual) {
+				actions_len -= 1;
+				continue;
+			}
```

Comment:
This should be checked when given an index too, not just next/prev. The switch_layout code also calls xkb_keymap_num_layouts, not just the code to get the current layout's index.
```suggestion
		if (action->keyboard->keymap == NULL && dev->is_virtual) {
			actions_len -= 1;
			continue;
		}
		if (relative) {
```

Reply:
You are right, I changed it as requested.